### DISCUSSION TITLE
Missing data for one hour issue

### DISCUSSION THREAD

**Discussion by kstobbe** [wmbusmeters/wmbusmeters#1263] [General]
Hi, I have a setup with at nanoCUL USB dongle connected to at Linux PC running Ubuntu LTS 22. The nanoCUL dongle has been FW updated to support a large buffer for C1 packages from a Kamstrup flowIQ 2200 water meter transmitting every 16th second. I have wmbusmeters running in a docker and it picks up wM-Bus and decodes it just fine, publishes it to a MQTT in another docker. The data is collected by a Telegraf instance and put into an InfluxDB. All is good, I get data almost every 16th second. 

BUT! Once in a while the data collection goes quiet for EXACTLY one hour. This happens 0-2 times a day. What puzzles me is that it is exactly one hour of silence.

I’ve looked in the wmbusmeters.log and there is not much help to find but a few hints about USB reset. But I think wmbusmeters will reset the USB regularly? I looked in journalctrl for hints about the USB and got suspicious about fwupd as it seemed to run and fail whenever I had a USB silence issue. But I’ve seen the issue after uninstalling fwupd so that turned out to be a cold lead. 

My suspicions are aimed at the wmbusmeters driver and handling of USB or the USB dongle FW itself or the general Linux USB handling. But I’m a bit at a loss on where to go from here. Suggestions are very welcome. 

Best regards, KS

**Comment by weetmuts** [comment]
wmbusmeters exercies the reset functionality every 23 hours by default. You can override this with --resetafter=0h (never reset)
or --resetafter=5m to reset every five minutes. The reset is not yet resetting the usb. It just repeates the dongle initilization after closing/reopening the filedescriptor. 

The reason is that some dongles do lock up after running for a couple of weeks. So, we exercies the reset, also the
reset code is pretty complicated, so  this also makes sure it is well tested. 

It does not sound like an exactly one-hour downtime has anything to do with this.....but you never know.

Anyway, set --resetafter=30m to exercies it more often, if you get more one-hour outages, then it might be connected.

**Reply by kstobbe** [reply]
I have the following from the wmbusmeters log - that must be the smoking gun wrt. the one hour. This seems like a kind of watchdog timer whereas `resetafter` sounds more like a fixed forced reset (which is also a good idea).

Is it possible to configure the inactivity watchdog timer? The dongle is in really good radio conditions and I'd like to reduce the timer to 300s - that leaves plenty of room to miss a few broadcasts, e.g. due to radio interference or similar, but is short enough to go unnoticed in my data analysis, whereas one hour is definitely not.

```
[ALARM DeviceInactivity] 3601 seconds of inactivity resetting /dev/ttyUSB0 cul (timeout 3600s expected mon-sun(00-23) now 2024-05-05 01:11)
(wmbus) resetting /dev/ttyUSB0:cul[]
(wmbus) reset completed /dev/ttyUSB0:cul[]
(wmbus) successfully reset wmbus device
[ALARM DeviceInactivity] 3601 seconds of inactivity resetting /dev/ttyUSB0 cul (timeout 3600s expected mon-sun(00-23) now 2024-05-05 07:04)
(wmbus) resetting /dev/ttyUSB0:cul[]
(wmbus) reset completed /dev/ttyUSB0:cul[]
(wmbus) successfully reset wmbus device
```

**Reply by weetmuts** [reply]
Yep you configure alarmtimeout=10h instead of 1h.

**Reply by weetmuts** [reply]
The alarmtimeout triggers because you do not receive anything for an hour. It should not affect the actual reception unless there is a bug where it continuosly reset for an hour.

**Reply by weetmuts** [reply]
It again is one of those workarounds to dongles that silently stop working.

**Reply by weetmuts** [reply]
Oh, if your silent dongle is fixed by the reset, then for sure reduce the timeout to something less!

**Reply by weetmuts** [reply]
alarmtimeout=5m
